build-support/bin/download_native_build_binaries.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Download the binaries required by cargo.sh to build our native code.
+# Sets associated env vars as needed.
+
+bin_cache="${HOME}/.cache/pants/bin"
+arch="$(uname)"
+
+if [[ ! "${arch}" =~ ^(Darwin|Linux)$ ]]; then
+    die "Unrecognized arch (uname=${arch})"
+fi
+
+arch_lower="$(echo "${arch}" | tr '[:upper:]' '[:lower:]')"
+
+# Download cmake
```

Comment:
It might still be a good idea to have a function to dedupe these calls.

Reply:
Thought of that, but there are a bunch of subtle differences in each case that made it not worth it, IMO.